backend/app/models/user.rb

```
@@ -17,30 +17,34 @@ class User < ApplicationRecord
+                u.provider = auth.provider
+                u.uid = auth.uid
+                u.role = "user"
+                u.username = auth.info.email&.split('@')&.first&.gsub(/[^a-zA-Z0-9_]/, '_') || "user_#{SecureRandom.hex(4)}"
+                end
+            end
+
+            # Ensure a user profile exists
+            UserProfile.find_or_create_by(user_id: user.id)
+
+            user
         end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `from_omniauth` is too high. [<8, 31, 8> 33/23]

(Metrics/AbcSize)